client/src/raw.rs

```diff
@@ -586,6 +609,9 @@ proxier! {
         |r| {
             let labels = namespaced_request!(r);
             r.extensions_mut().insert(labels);
+            if r.get_ref().wait_new_event {
+                r.extensions_mut().insert(ForceConsiderLongPoll);
+            }
```

Comment:
I wonder if this should also occur on `UpdateWorkflowExecution` and `PollWorkflowExecutionUpdate`

Reply:
I considered it, but update pretty sure you wouldn't want to work exactly the same way where it retries forever. Poll could be ok to do that with potentially.

Reply:
Hrmm, arguably we wouldn't want that for get history either. Users may wait for next history event and wait for update response the same way.